scripts/o3de/ExportScripts/export_source_android.py

```diff
@@ -0,0 +1,383 @@
+                                        auto_key_pwd:str|None = None):
+    is_pc = exp.get_default_asset_platform() == 'pc'
+    android_sdk_home_env_var = '%ANDROID_SDK_HOME%' if is_pc else '$ANDROID_SDK_HOME'
+    android_signing_config_keystore_file_env_var = '%ANDROID_SIGNING_CONFIG_KEYSTORE_FILE%' if is_pc else '$ANDROID_SIGNING_CONFIG_KEYSTORE_FILE' 
+    android_signing_config_key_alias_env_var = '%ANDROID_SIGNING_CONFIG_KEY_ALIAS%' if is_pc else '$ANDROID_SIGNING_CONFIG_KEY_ALIAS'
+
+    android_global_config, _ = android.get_android_config(is_global=True)
+    android_global_config.set_config_value(key=android_support.SETTINGS_SDK_ROOT.key, value=android_sdk_home_env_var, validate_value=False)
+    android_global_config.set_config_value(key=android_support.SETTINGS_PLATFORM_SDK_API.key, value=api, validate_value=False)
+    android_global_config.set_config_value(key=android_support.SETTINGS_NDK_VERSION.key, value=ndk_version, validate_value=False)
+    android_global_config.set_config_value(key=android_support.SETTINGS_GRADLE_PLUGIN_VERSION.key, value=gradle_min_version, validate_value=False)
+    android_global_config.set_config_value(key=android_support.SETTINGS_SIGNING_STORE_FILE.key, value=android_signing_config_keystore_file_env_var, validate_value=False)
+    android_global_config.set_config_value(key=android_support.SETTINGS_SIGNING_KEY_ALIAS.key, value=android_signing_config_key_alias_env_var, validate_value=False)
+
+    if not auto_store_pwd:
+        android_global_config.set_password(android_support.SETTINGS_SIGNING_STORE_PASSWORD.key)
+    else:
+        android_global_config.set_config_value(key=android_support.SETTINGS_SIGNING_STORE_PASSWORD.key, value=auto_store_pwd, validate_value=False)
+    
+    if not auto_key_pwd:
+        android_global_config.set_password(android_support.SETTINGS_SIGNING_KEY_PASSWORD.key)
+    else:
+        android_global_config.set_config_value(key=android_support.SETTINGS_SIGNING_KEY_PASSWORD.key, value=auto_key_pwd, validate_value=False)
+
+def deloy_to_android_device(target_android_project_path: pathlib.Path,
+                            build_config:str,
+                            project_name:str,
+                            android_sdk_home:pathlib.Path,
+                            org_name:str|None = None,
+                            activity_name:str|None = None):
+    is_pc = exp.get_default_asset_platform() == 'pc'
+
+    adb_exec_path = android_sdk_home / ('platform-tools/adb' + ('.exe' if is_pc else ''))
+    exp.process_command([adb_exec_path, 'install', '-t', '-r', target_android_project_path / f'app/build/outputs/apk/{build_config}/app-{build_config}.apk'])
+
+    time.sleep(1)
+
+    if not activity_name:
+        activity_name = f'{project_name}Activity'
+    
+    if not org_name:
+        org_name = f'org.o3de.{project_name}'
+
+    exp.process_command([adb_exec_path, 'shell', 'am', 'start', '-a', 'android.intent.action.MAIN','-n', f'{org_name}/.{activity_name}'])
```

Comment:
It looks like this call returns the exit code. This needs to make sure the exit code is success and raise an error if it is not